src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -453,6 +453,63 @@ public function isAbsolutePath($file)
     }
 
     /**
+     * Creates a temporary file with support for custom stream wrappers.
+     *
+     * @param string $dir    The directory where the temporary filename will be created.
+     * @param string $prefix The prefix of the generated temporary filename.
+     *                       Note: Windows uses only the first three characters of prefix.
+     *
+     * @return string The new temporary filename (with path), or false on failure.
+     */
+    public function tempNam($dir, $prefix)
+    {
+        $limit = 10;
+        $schemeSeparator = '://';
+        $scheme = $this->getScheme($dir);
+        $hierarchy = $this->getHierarchy($dir);
```

Comment:
rather than implementing these in 2 separate methods, each of them redoing the same splitting of the path, what about doing the logic determining both in the same place (either here or in a method returning ``array($scheme, $hierarchy)`` and using ``list($scheme, $hierarchy) = ...`` here) ?